models/demos/bert_tiny/tests/perf_e2e_bert_tiny.py

```diff
@@ -0,0 +1,320 @@
+    # test_infra.input_tensor = ttnn.to_memory_config(tt_input_ids, input_mem_config)
+    test_infra.input_tensor = tt_input_ids  # ttnn.to_memory_config(tt_input_ids, input_mem_config)
+    test_infra.token_type_ids = tt_token_type_ids
+    test_infra.position_ids = tt_position_ids
+    test_infra.attention_mask = tt_attention_mask
+    ttnn.record_event(0, op_event)
+    test_infra.output_tensor.deallocate(force=True)
+    trace_input_addr = ttnn.buffer_address(test_infra.input_tensor)
+
+    tid = ttnn.begin_trace_capture(device, cq_id=0)
+    tt_output_res = test_infra.run()
+    # input_tensor = ttnn.allocate_tensor_on_device(
+    #     shape,
+    #     dtype,
+    #     layout,
+    #     device,
+    #     input_mem_config,
+    # )
+    ttnn.end_trace_capture(device, tid, cq_id=0)
+    # assert trace_input_addr == ttnn.buffer_address(input_tensor)
+    ttnn.dump_device_profiler(device)
+
+    for iter in range(0, num_warmup_iterations):
+        ttnn.wait_for_event(1, op_event)
+        # ttnn.copy_host_to_device_tensor(tt_inputs_host, tt_input_ids, 1)
+        ttnn.copy_host_to_device_tensor(tt_inputs_host, tt_input_ids, 1)
+        ttnn.copy_host_to_device_tensor(tt_token_type_ids_host, tt_token_type_ids, 1)
+        ttnn.copy_host_to_device_tensor(tt_position_ids_host, tt_position_ids, 1)
+        ttnn.copy_host_to_device_tensor(tt_attention_mask_host, tt_attention_mask, 1)
+        ttnn.record_event(1, write_event)
+        ttnn.wait_for_event(0, write_event)
+        # input_tensor = ttnn.reshard(tt_input_ids, input_mem_config, input_tensor)
+        ttnn.record_event(0, op_event)
+        ttnn.execute_trace(device, tid, cq_id=0, blocking=True)
+        ttnn.dump_device_profiler(device)
+
+    ttnn.synchronize_devices(device)
+    if use_signpost:
+        signpost(header="start")
+    outputs = []
+    profiler.start(f"run")
+    for iter in range(0, num_measurement_iterations):
+        ttnn.wait_for_event(1, op_event)
+        # ttnn.copy_host_to_device_tensor(tt_inputs_host, tt_input_ids, 1)
+        ttnn.copy_host_to_device_tensor(tt_inputs_host, tt_input_ids, 1)
+        ttnn.copy_host_to_device_tensor(tt_token_type_ids_host, tt_token_type_ids, 1)
+        ttnn.copy_host_to_device_tensor(tt_position_ids_host, tt_position_ids, 1)
+        ttnn.copy_host_to_device_tensor(tt_attention_mask_host, tt_attention_mask, 1)
+        ttnn.record_event(1, write_event)
+        ttnn.wait_for_event(0, write_event)
+        # TODO: Add in place support to ttnn to_memory_config
+        # input_tensor = ttnn.reshard(tt_input_ids, input_mem_config, input_tensor)
+        ttnn.record_event(0, op_event)
+        ttnn.execute_trace(device, tid, cq_id=0, blocking=False)
+        outputs.append(tt_output_res.cpu(blocking=False))
+    ttnn.synchronize_devices(device)
+    profiler.end(f"run")
+    if use_signpost:
+        signpost(header="stop")
+    ttnn.dump_device_profiler(device)
+
+    ttnn.release_trace(device, tid)
```

Comment:
You should probably have some asserts (like checking PCC) in here to make sure things are working as expected.